crates/noirc_evaluator/src/ssa/ir/dfg.rs

```diff
@@ -149,7 +156,7 @@ impl DataFlowGraph {
         instruction: Instruction,
         block: BasicBlockId,
         ctrl_typevars: Option<Vec<Type>>,
-        location: Option<Location>,
+        call_stack: CallStack,
```

Comment:
It would be a compiler error if callstack was empty here right?

Reply:
No, we allow empty call stacks for e.g. testing or for extra instructions SSA passes may insert that were not directly present in the source code. The defunctionalization pass for example creates `apply` functions in which none of the instructions have locations or call stacks.